Assets/Scripts/UI/PopupWindow.cs

```diff
@@ -0,0 +1,166 @@
+using System.Collections.Generic;
+using Events;
+using UnityEngine;
+using UnityEngine.UI;
+
+namespace UI
+{
+    public class PopupWindow : MonoBehaviour
+    {
+        [SerializeField] private List<GameObject> popupObjects;
+        [SerializeField] private List<Image> itemSprites;
+        [SerializeField] private List<Sprite> itemLargeSprites;
+
+        [SerializeField] private Image largeImage;
+
+        private List<Outline> _itemOutlines = new List<Outline>();
+        private int _selectIndex = 0;
+
+        [SerializeField] private bool isOpen = false;
+
+        private float _targetSize = 0f;
+        private float _largeTargetSize = 0f;
+
+        [SerializeField] [Header("Popup Item")]
+        private bool popupHasItem;
+
+        [SerializeField] private int itemPosition;
+        [SerializeField] private Sprite substituteSprite;
+        [SerializeField] private GameObject popupItem;
+        [SerializeField] private DefaultEvent eventToTrigger;
+        private bool _hasGottenItem = false;
+
+        private void Awake()
+        {
+            itemPosition--;
+
+            foreach (GameObject item in popupObjects)
+            {
+                _itemOutlines.Add(item.GetComponent<Outline>());
+            }
+
+            ResetSpriteColor();
+        }
+
+        public void OpenClosePopup()
+        {
+            switch (isOpen)
+            {
+                case true:
+                    _targetSize = 0f;
+                    break;
+
+                case false:
+                    _targetSize = 1f;
+                    ResetSelection();
+                    break;
+            }
+
+            isOpen = !isOpen;
+        }
+
+        private void Update()
+        {
+            transform.localScale = Vector3.Lerp(transform.localScale,
+                new Vector3(_targetSize, _targetSize, _targetSize), Time.deltaTime * 5f);
+            largeImage.transform.localScale = Vector3.Lerp(largeImage.transform.localScale,
+                new Vector3(_largeTargetSize, _largeTargetSize, _largeTargetSize), Time.deltaTime * 12.5f);
+
+            if (Input.GetKeyDown(KeyCode.LeftArrow) && isOpen)
+            {
+                SelectPrevious();
+            }
+
+            if (Input.GetKeyDown(KeyCode.RightArrow) && isOpen)
+            {
+                SelectNext();
+            }
+
+            if (Input.GetKeyDown(KeyCode.UpArrow) && isOpen)
```

Comment:
This needs to be hooked up to our input system. I'm adding enum on CharacterHandler for seeing what input mode the ghost or human is in then they don't apply moving input and jump input.